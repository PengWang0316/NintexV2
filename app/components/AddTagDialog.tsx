import React, {
  useState, memo, ReactElement, ChangeEvent,
} from 'react';
import { connect } from 'react-redux';
import {
  Button, Dialog, DialogActions, DialogContent,
  DialogContentText, DialogTitle, Chip, TextField,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import I18n from '@kevinwang0316/i18n';
import { CirclePicker } from 'react-color';

import { addTag as addTagAction } from '../store/Tags/actions';
import { TagsType } from '../store/Tags/types';
import { AppState } from '../store/ConfigureStore';

interface Props {
  open: boolean;
  tags: TagsType | null;
  addTag: (content: string, color: string) => void;
  handleClose: (event: React.MouseEvent) => void;
}

const useStyles = makeStyles({
  flexDiv: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginTop: 20,
  },
  secondContentText: { marginTop: 30 },
  btnDiv: { textAlign: 'right' },
});

const DEFAULT_COLOT = '#f44336';
const COLOR_REGEXP = /^#[\d\w]{6}$/;

export const AddTagDialog = ({
  open, handleClose, tags = null, addTag,
}: Props): ReactElement => {
  const classes = useStyles({});
  const [tagColor, setTagColor] = useState(DEFAULT_COLOT);
  const [tagText, setTagText] = useState('');

  const handleChangeComplete = color => setTagColor(color.hex);

  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => setTagText(event.target.value);

  const handleAddBtnClick = () => {
    // Do a validation here and also do it in the backend again
    if (tagText && tagText !== '' && tagColor.match(COLOR_REGEXP)) {
      addTag(tagText, tagColor);
      setTagText('');
      setTagColor(DEFAULT_COLOT);
    }
  };

  return (
    <Dialog
      open={open}
      keepMounted
      maxWidth="md"
      onClose={handleClose}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">{I18n.get('addTagDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {I18n.get('addTagDialogExist')}
        </DialogContentText>
        {tags && (
          <div>
            {Object.keys(tags).map(tag => (
              <Chip
                label={`${tags[tag][0]}`}
                key={`${tags[tag][0]}${tags[tag][1]}`}
                clickable
                style={{ backgroundColor: `${tags[tag][1]}`, margin: '0 10px 5px 0', color: 'white' }}
              />
            ))}
          </div>
        )}
        <DialogContentText className={classes.secondContentText}>
          {I18n.get('addTagDialogContent')}
        </DialogContentText>
        <div className={classes.flexDiv}>
          <div>
            <div>
              <TextField
                label={I18n.get('tagContentTitle')}
                value={tagText}
              // className={classes.textField}
                margin="normal"
                onChange={handleInputChange}
              />
            </div>
            <div className={classes.btnDiv}>
              <Button size="small" variant="outlined" onClick={handleAddBtnClick} color="secondary">
                {I18n.get('add')}
              </Button>
            </div>

          </div>
          <div><CirclePicker onChangeComplete={handleChangeComplete} color={tagColor} /></div>
        </div>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('close')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};

const mapStateToProps = (state: AppState) => ({
  tags: state.tags,
});
const mapDispatchToProps = dispatch => ({
  addTag: (content: string, color: string) => dispatch(addTagAction(content, color)),
});
export default connect(mapStateToProps, mapDispatchToProps)(memo(AddTagDialog));
