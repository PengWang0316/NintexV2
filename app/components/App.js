import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import importedComponent from 'react-imported-component';
import CssBaseline from '@material-ui/core/CssBaseline';
import Amplify from 'aws-amplify';
import { withAuthenticator } from 'aws-amplify-react';

import Navbar from './Navbar';
import MenuDrawer from './MenuDrawer';
import { checkInstanceStatus as checkInstanceStatusAction } from '../actions/InstanceActions';
import {
  HOME_PAGE_URL, WORKFLOW_MANAGER_PAGE_URL, SIGNIN_PAGE_URL,
  cognitoConfig, amplifyAuthSignOption, MONITOR_CENTER_PAGE_URL,
} from '../config';
import LoadingAnimation from './SharedComponents/LoadingAnimation';

Amplify.configure({
  Auth: cognitoConfig,
});

const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
  palette: {
    primary: {
      light: '#4A6572',
      main: '#344955',
      dark: '#232F34',
      contrastText: '#fff',
    },
    secondary: {
      light: '#497ca4',
      main: '#105075',
      dark: '#002884',
      contrastText: '#fff',
    },
  },
});

/* istanbul ignore next */
const HomePage = importedComponent(() => import(/* webpackChunkName: "HomePageContainer" *//* webpackPrefetch: true */ './containers/HomePageContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });
const ManagementPage = importedComponent(() => import(/* webpackChunkName: "ManagementContainer" *//* webpackPrefetch: true */ './containers/ManagementContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });
const MonitorPage = importedComponent(() => import(/* webpackChunkName: "MonitorContainer" *//* webpackPrefetch: true */ './containers/MonitorContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });

const INTERVAL_TIME = 5 * 60 * 1000;
let autoMonitorJob = null;

/**
 * The root component that contains the theme, routers, navbar, and login dialog
 */
export const App = ({ monitorList, checkInstanceStatus }) => {
  useEffect(() => {
    if (Object.keys(monitorList.data).length !== 0) {
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = setInterval(() => checkInstanceStatus(monitorList.data), INTERVAL_TIME);
      // console.log('item changed, start a new job');
    } else if (autoMonitorJob && Object.keys(monitorList.data).length === 0) {
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = null;
      // console.log('empty list, stop job');
    }
  });
  return (
    <MuiThemeProvider theme={theme}>
      <Router>
        <div>
          <CssBaseline />
          <Navbar />
          <MenuDrawer />
          <main>
            <Switch>
              <Route exact path={HOME_PAGE_URL} component={HomePage} />
              <Route exact path={WORKFLOW_MANAGER_PAGE_URL} component={ManagementPage} />
              <Route exact path={MONITOR_CENTER_PAGE_URL} component={MonitorPage} />
              <Route exact path={SIGNIN_PAGE_URL} component={withAuthenticator(HomePage, amplifyAuthSignOption)} />
              <Route render={() => <p>Not Fount!</p>} />
            </Switch>
          </main>
        </div>
      </Router>
    </MuiThemeProvider>
  );
};
App.propTypes = {
  monitorList: PropTypes.objectOf(PropTypes.any).isRequired,
  checkInstanceStatus: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({ ...state });
const mapDispatchToProps = {
  checkInstanceStatus: checkInstanceStatusAction,
};
export default connect(mapStateToProps, mapDispatchToProps)(App);
