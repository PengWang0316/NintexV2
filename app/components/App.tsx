import React, { Suspense, lazy } from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import Amplify from 'aws-amplify';
import { withAuthenticator } from 'aws-amplify-react';

import Navbar from './Navbar';
import MenuDrawer from './MenuDrawer';
import { checkInstanceStatus as checkInstanceStatusAction } from '../store/MonitorList/actions';
import {
  HOME_PAGE_URL, WORKFLOW_MANAGER_PAGE_URL, SIGNIN_PAGE_URL,
  cognitoConfig, amplifyAuthSignOption, MONITOR_CENTER_PAGE_URL,
} from '../config';
import { MonitorListType, MonitorListData } from '../store/MonitorList/types';

Amplify.configure({
  Auth: cognitoConfig,
});

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#4A6572',
      main: '#344955',
      dark: '#232F34',
      contrastText: '#fff',
    },
    secondary: {
      light: '#497ca4',
      main: '#105075',
      dark: '#002884',
      contrastText: '#fff',
    },
  },
});

/* istanbul ignore next */
const HomePage = lazy(() => import(/* webpackChunkName: "HomePageContainer" *//* webpackPrefetch: true */ './containers/HomePageContainer'));
/* istanbul ignore next */
const ManagementPage = lazy(() => import(/* webpackChunkName: "ManagementContainer" *//* webpackPrefetch: true */ './containers/ManagementContainer'));
/* istanbul ignore next */
const MonitorPage = lazy(() => import(/* webpackChunkName: "MonitorContainer" *//* webpackPrefetch: true */ './containers/MonitorContainer'));

const INTERVAL_TIME = 1 * 60 * 1000;
let autoMonitorJob = null;
interface PropType {
  monitorList: MonitorListType;
  checkInstanceStatus: (monitoredWorkflows: MonitorListData) => void;
}
/**
 * The root component that contains the theme, routers, navbar, and login dialog
 */
export const App = ({ monitorList, checkInstanceStatus }: PropType) => {
  React.useEffect(() => {
    if (Object.keys(monitorList.data).length !== 0) {
      // item changed, start a new job
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = setInterval(() => checkInstanceStatus(monitorList.data), INTERVAL_TIME);
    } else if (autoMonitorJob && Object.keys(monitorList.data).length === 0) {
      // Stop the fetching job
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = null;
    }
  });
  return (
    <MuiThemeProvider theme={theme}>
      <Router>
        <div>
          <CssBaseline />
          <Navbar />
          <MenuDrawer />
          <main>
            <Suspense fallback={<div>Loading...</div>}>
              <Switch>
                <Route exact path={HOME_PAGE_URL} component={HomePage} />
                <Route exact path={WORKFLOW_MANAGER_PAGE_URL} component={ManagementPage} />
                <Route exact path={MONITOR_CENTER_PAGE_URL} component={MonitorPage} />
                <Route exact path={SIGNIN_PAGE_URL} component={withAuthenticator(HomePage, amplifyAuthSignOption)} />
                <Route render={() => <p>Not Fount!</p>} />
              </Switch>
            </Suspense>
          </main>
        </div>
      </Router>
    </MuiThemeProvider>
  );
};
/* istanbul ignore next */
const mapStateToProps = ({ monitorList }: { monitorList: MonitorListType }) => ({ monitorList });
/* istanbul ignore next */
const mapDispatchToProps = {
  checkInstanceStatus: checkInstanceStatusAction,
};
export default connect(mapStateToProps, mapDispatchToProps)(App);
