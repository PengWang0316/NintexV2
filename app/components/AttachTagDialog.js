import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Slide, Chip,
} from '@material-ui/core';
import I18n from '@kevinwang0316/i18n';

import { updateTagFromWorkflow as updateTagFromWorkflowAction } from '../store/Workflows/actions';

const Transition = React.forwardRef((props, ref) => <Slide direction="down" ref={ref} {...props} />);

export const AttachTagDialog = ({
  isOpen, handleClose, tags, workflowId, workflows, updateTagFromWorkflow,
}) => {
  const existedTagIds = Object.keys(workflows.data).length !== 0 && workflowId && workflows.data[workflowId].tags ? workflows.data[workflowId].tags.split(',') : [];
  const handleTagClick = event => updateTagFromWorkflow(
    workflowId,
    `${workflows.data[workflowId].tags ? `${workflows.data[workflowId].tags},` : ''}${event.target.tagName === 'span' ? event.target.parentNode.parentNode.getAttribute('name') : event.target.parentNode.getAttribute('name')}`,
  );


  return (
    <Dialog
      open={isOpen}
      TransitionComponent={Transition}
      keepMounted
      maxWidth="md"
      onClose={handleClose}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">{I18n.get('addTagDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {I18n.get('attachTagDialogContent')}
        </DialogContentText>
        {tags && workflows && (
          <div>
            {Object.keys(tags).filter(tagId => !existedTagIds.includes(tagId)).map(tag => (
              <Chip
                label={`${tags[tag][0]}`}
                key={`${tags[tag][0]}${tags[tag][1]}`}
                name={`${tag}`}
                clickable
                onClick={handleTagClick}
                style={{ backgroundColor: `${tags[tag][1]}`, margin: '0 10px 5px 0', color: 'white' }}
              />
            ))}
          </div>
        )}
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('close')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
AttachTagDialog.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  workflowId: PropTypes.string,
  workflows: PropTypes.objectOf(PropTypes.any),
  handleClose: PropTypes.func.isRequired,
  updateTagFromWorkflow: PropTypes.func.isRequired,
  tags: PropTypes.objectOf(PropTypes.array),
};
AttachTagDialog.defaultProps = { tags: null, workflowId: null, workflows: null };
const mapStateToProps = state => ({
  tags: state.tags,
  workflows: state.workflows,
});
const mapDispatchToProps = dispatch => ({
  updateTagFromWorkflow:
    (workflowId, tagIds) => dispatch(updateTagFromWorkflowAction(workflowId, tagIds)),
});
export default connect(mapStateToProps, mapDispatchToProps)(AttachTagDialog);
