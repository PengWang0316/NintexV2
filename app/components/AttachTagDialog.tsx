import React, { memo, useMemo, useCallback } from 'react';
import { connect } from 'react-redux';
import {
  Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Slide, Chip,
} from '@material-ui/core';
import { ChipProps } from '@material-ui/core/Chip';
import I18n from '@kevinwang0316/i18n';

import { updateTagFromWorkflow as updateTagFromWorkflowAction } from '../store/Workflows/actions';
import { TagsType } from '../store/Tags/types';
import { Workflows } from '../store/Workflows/types';
import { AppState } from '../store/ConfigureStore';

// Use a customized interface to add name prop to the Chip
interface CustomizedChipType extends ChipProps {
  name: string;
}

// Defind a new wrapper for the Chip component
const CustomizedChip: React.ElementType = (props: CustomizedChipType) => <Chip {...props} />;


interface Props {
  isOpen: boolean;
  tags: TagsType;
  workflowId: string;
  workflows: Workflows;
  handleClose: (event: React.MouseEvent) => void;
  updateTagFromWorkflow: (workflowId: string, tagIds: string) => void;
}

const Transition = React.forwardRef((props, ref) => <Slide direction="down" ref={ref} {...props} />);

export const AttachTagDialog = ({
  isOpen, handleClose, tags = null, workflowId = null, workflows = null, updateTagFromWorkflow,
}: Props) => {
  const existedTagIds = useMemo(() => (Object.keys(workflows.data).length !== 0 && workflowId && workflows.data[workflowId].tags ? workflows.data[workflowId].tags.split(',') : []), [workflows.data]);
  const handleTagClick = useCallback((event: React.MouseEvent) => {
    const target: HTMLElement = event.target as HTMLElement;
    updateTagFromWorkflow(
      workflowId,
      `${workflows.data[workflowId].tags ? `${workflows.data[workflowId].tags},` : ''}${target.tagName === 'span' ? (target.parentNode.parentNode as HTMLElement).getAttribute('name') : (target.parentNode as HTMLElement).getAttribute('name')}`,
    );
  }, [workflows.data]);

  return (
    <Dialog
      open={isOpen}
      TransitionComponent={Transition as React.ComponentType}
      keepMounted
      maxWidth="md"
      onClose={handleClose}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">{I18n.get('addTagDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {I18n.get('attachTagDialogContent')}
        </DialogContentText>
        {tags && workflows && (
          <div>
            {Object.keys(tags).filter(tagId => !existedTagIds.includes(tagId)).map(tag => (
              <CustomizedChip
                label={`${tags[tag][0]}`}
                key={`${tags[tag][0]}${tags[tag][1]}`}
                name={`${tag}`}
                clickable
                onClick={handleTagClick}
                style={{ backgroundColor: `${tags[tag][1]}`, margin: '0 10px 5px 0', color: 'white' }}
              />
            ))}
          </div>
        )}
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('close')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
const mapStateToProps = (state: AppState) => ({
  tags: state.tags,
  workflows: state.workflows,
});
const mapDispatchToProps = dispatch => ({
  updateTagFromWorkflow: (
    workflowId: string, tagIds: string,
  ) => dispatch(updateTagFromWorkflowAction(workflowId, tagIds)),
});
export default connect(mapStateToProps, mapDispatchToProps)(memo(AttachTagDialog));
