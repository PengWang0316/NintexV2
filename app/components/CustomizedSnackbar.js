import React from 'react';
import PropTypes from 'prop-types';
import { Close as CloseIcon, CheckCircle as CheckCircleIcon } from '@material-ui/icons';
import {
  IconButton, Fade, Snackbar, SnackbarContent,
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  icon: {
    fontSize: 20,
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing(1),
  },
  message: {
    display: 'flex',
    alignItems: 'center',
  },
});

export const CustomizedSnackbar = ({
  open, handleClose, backgroundColor, content, duration, classes,
}) => (
  <Snackbar
    open={open}
    onClose={handleClose}
    TransitionComponent={Fade}
    autoHideDuration={duration}
  >
    <SnackbarContent
      style={{ backgroundColor }}
      message={(
        <span className={classes.message}>
          <CheckCircleIcon className={classes.iconVariant} />
          {content}
        </span>
      )}
      action={[
        <IconButton
          key="close"
          color="inherit"
          onClick={handleClose}
        >
          <CloseIcon className={classes.icon} />
        </IconButton>,
      ]}
    />
  </Snackbar>
);
CustomizedSnackbar.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  duration: PropTypes.number,
};
CustomizedSnackbar.defaultProps = { duration: 3000 };
export default withStyles(styles)(CustomizedSnackbar);
