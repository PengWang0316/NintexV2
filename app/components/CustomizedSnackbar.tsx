import React, { ReactElement } from 'react';
import { Close as CloseIcon, CheckCircle as CheckCircleIcon } from '@material-ui/icons';
import {
  IconButton, Fade, Snackbar, SnackbarContent,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

interface Props {
  open: boolean;
  handleClose: (event: React.MouseEvent) => void;
  backgroundColor: string;
  content: string;
  duration?: number;
}

const useStyles = makeStyles(theme => ({
  icon: {
    fontSize: 20,
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing(1),
  },
  message: {
    display: 'flex',
    alignItems: 'center',
  },
}));

export const CustomizedSnackbar = ({
  open, handleClose, backgroundColor, content, duration = 3000,
}: Props): ReactElement => {
  const classes = useStyles({});
  return (
    <Snackbar
      open={open}
      onClose={handleClose}
      TransitionComponent={Fade}
      autoHideDuration={duration}
    >
      <SnackbarContent
        style={{ backgroundColor }}
        message={(
          <span className={classes.message}>
            <CheckCircleIcon className={classes.iconVariant} />
            {content}
          </span>
        )}
        action={[
          <IconButton
            key="close"
            color="inherit"
            onClick={handleClose}
          >
            <CloseIcon className={classes.icon} />
          </IconButton>,
        ]}
      />
    </Snackbar>
  );
};

export default CustomizedSnackbar;
