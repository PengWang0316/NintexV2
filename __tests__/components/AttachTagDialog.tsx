import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { AttachTagDialog } from '../../app/components/AttachTagDialog';

jest.mock('@material-ui/core/Button', () => 'Button');
jest.mock('@material-ui/core/Dialog', () => 'Dialog');
jest.mock('@material-ui/core/DialogActions', () => 'DialogActions');
jest.mock('@material-ui/core/DialogContent', () => 'DialogContent');
jest.mock('@material-ui/core/DialogContentText', () => 'DialogContentText');
jest.mock('@material-ui/core/DialogTitle', () => 'DialogTitle');
jest.mock('@material-ui/core/Slide', () => 'Slide');
jest.mock('@material-ui/core/Chip', () => 'Chip');
jest.mock('@material-ui/core/Button', () => 'Button');
jest.mock('@kevinwang0316/i18n', () => ({ get: jest.fn().mockImplementation(str => str) }));

describe('Test AttachTagDialog', () => {
  const defaultProps = {
    isOpen: false,
    tags: null,
    workflowId: 'workflowId',
    workflows: { isFetched: false, data: {} },
    handleClose: jest.fn(),
    updateTagFromWorkflow: jest.fn(),
  };

  const tags = {
    1: ['IT', '#e91e63'],
    2: ['Accounting', '#03a9f4'],
    3: ['Marketing', '#607d8b'],
    4: ['SDE', '#ff5722'],
  };

  const workflows = {
    isFetched: true,
    data: {
      '006D7B35-20A7-E811-BCE7-000D3A320C2B': {
        workflowId: '006D7B35-20A7-E811-BCE7-000D3A320C2B',
        workflowName: 'ISV Order Form - Prod',
        publishDate: '2018-11-27T00:00:00.000Z',
        publisher: 'Eric',
        locationPath: 'NWC / nintex-it (Eric Harris)',
        tags: '2,3,4',
      },
      '007600E3-69BB-E711-80C2-000D3A32028C': {
        workflowId: '007600E3-69BB-E711-80C2-000D3A32028C',
        workflowName: 'LazyApprovalTest',
        publishDate: '2017-10-27T00:00:00.000Z',
        publisher: 'NTXSupport Administrator',
        locationPath: 'SP2013-1 / Home / Demo',
        tags: '2,1',
      },
    },
  };

  beforeEach(() => jest.clearAllMocks());

  it('Should renderd correctly without CustomizedChip component', () => {
    const component = shallow(<AttachTagDialog {...defaultProps} />);

    expect(component.exists('Dialog')).toBe(true);

    const DialogTitle = component.find('DialogTitle');
    expect(DialogTitle.text()).toEqual('addTagDialogTitle');

    expect(component.exists('DialogContent')).toBe(true);

    const DialogContentText = component.find('DialogContentText');
    expect(DialogContentText.text()).toEqual('attachTagDialogContent');
    expect(DialogContentText.prop('id')).toEqual('alert-dialog-slide-description');

    expect(component.exists('DialogActions')).toBe(true);

    const Button = component.find('Button');
    expect(Button.prop('color')).toEqual('primary');
    expect(Button.text()).toEqual('close');

    expect(component.exists('CustomizedChip')).toBe(false);
  });

  it('Should have one CustomizedChip', () => {
    const props = {
      ...defaultProps, tags, workflows, workflowId: '006D7B35-20A7-E811-BCE7-000D3A320C2B',
    };
    // @ts-ignore
    const component = shallow(<AttachTagDialog {...props} />);

    const tagsComponent = component.find('CustomizedChip');
    expect(tagsComponent.length).toBe(1);
  });

  it('Should have two CustomizedChips', () => {
    const props = {
      ...defaultProps, tags, workflows, workflowId: '007600E3-69BB-E711-80C2-000D3A32028C',
    };
    // @ts-ignore
    const component = shallow(<AttachTagDialog {...props} />);

    const tagsComponent = component.find('CustomizedChip');
    expect(tagsComponent.length).toBe(2);
  });

  it('Should call updateTagFromWorkflow', () => {
    const props = {
      ...defaultProps, tags, workflows, workflowId: '006D7B35-20A7-E811-BCE7-000D3A320C2B',
    };
    // @ts-ignore
    const component = shallow(<AttachTagDialog {...props} />);

    const tagsComponent = component.find('CustomizedChip');
    tagsComponent.simulate('click', { target: { tagName: 'span', parentNode: { parentNode: { getAttribute: jest.fn() } } } });

    expect(defaultProps.updateTagFromWorkflow).toHaveBeenCalledTimes(1);
  });

  it('Should call handlClose', () => {
    const component = shallow(<AttachTagDialog {...defaultProps} />);
    component.find('Button').simulate('click');

    expect(defaultProps.handleClose).toHaveBeenCalledTimes(1);
  });

  test('Snapshot test for no CustomizedChip', () => expect(renderer.create(<AttachTagDialog {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('Snapshot test for having two CustomizedChip', () => {
    // @ts-ignore
    expect(renderer.create(<AttachTagDialog {...{
      ...defaultProps, tags, workflows, workflowId: '007600E3-69BB-E711-80C2-000D3A32028C',
    }}
    />).toJSON()).toMatchSnapshot();
  });
});
