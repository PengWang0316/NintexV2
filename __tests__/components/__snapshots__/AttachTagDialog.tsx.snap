// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AttachTagDialog Snapshot test for having two CustomizedChip 1`] = `
<Dialog
  TransitionComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  aria-describedby="alert-dialog-slide-description"
  aria-labelledby="alert-dialog-slide-title"
  keepMounted={true}
  maxWidth="md"
  onClose={[MockFunction]}
  open={false}
>
  <DialogTitle
    id="alert-dialog-slide-title"
  >
    addTagDialogTitle
  </DialogTitle>
  <DialogContent>
    <DialogContentText
      id="alert-dialog-slide-description"
    >
      attachTagDialogContent
    </DialogContentText>
    <div>
      <Chip
        clickable={true}
        label="Marketing"
        name="3"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#607d8b",
            "color": "white",
            "margin": "0 10px 5px 0",
          }
        }
      />
      <Chip
        clickable={true}
        label="SDE"
        name="4"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#ff5722",
            "color": "white",
            "margin": "0 10px 5px 0",
          }
        }
      />
    </div>
  </DialogContent>
  <DialogActions>
    <Button
      color="primary"
      onClick={[MockFunction]}
    >
      close
    </Button>
  </DialogActions>
</Dialog>
`;

exports[`Test AttachTagDialog Snapshot test for no CustomizedChip 1`] = `
<Dialog
  TransitionComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  aria-describedby="alert-dialog-slide-description"
  aria-labelledby="alert-dialog-slide-title"
  keepMounted={true}
  maxWidth="md"
  onClose={[MockFunction]}
  open={false}
>
  <DialogTitle
    id="alert-dialog-slide-title"
  >
    addTagDialogTitle
  </DialogTitle>
  <DialogContent>
    <DialogContentText
      id="alert-dialog-slide-description"
    >
      attachTagDialogContent
    </DialogContentText>
  </DialogContent>
  <DialogActions>
    <Button
      color="primary"
      onClick={[MockFunction]}
    >
      close
    </Button>
  </DialogActions>
</Dialog>
`;
